### DISCUSSION TITLE
How to do video editing on ImageMagick

### DISCUSSION THREAD

**Discussion by satimis** [ImageMagick/ImageMagick#5935] [Digital Image Processing]
Hi all,

How to do video editing on ImageMagick, running commands?

I have a .pdf image file created on >60images.  I still retain all original images.  I expect to create a slideshow direct on the .pdf image file, adding background music, subtitile, narration etc.

Can I do it on ImgaeMagick ?

Please advise.  Thanks

Regards

**Comment by snibgo** [comment]
IM is a raster image processor. Yes, it can be used as a component of a video editor. And if your requirements are very simple, it might be the only component (aside from ffmpeg as a delegate), or it can be called from a script or compiled program.

However, there are many (free, open-source) video editors available that might be easier for you.

**Comment by satimis** [comment]
Hi,
Thanks for your advice.
`However, there are many (free, open-source) video editors available that might be easier for you.`
You're right.
I have no problem running OpenShot, a free open source Video Editor, creating slideshow on *jpg, *png files, adding background music, subtitles, narration etc and export it as .mp4 file

Just for curiosity.  Whether It can be done on ImageMagicks ?

Regards

**Reply by fmw42** [reply]
Imagemagick has no direct ability to add audio as far as I know. You can add titles to frames and do other things (morphing and fades) and write to mpeg via ffmpeg or mp4 video.

**Comment by satimis** [comment]
Hi,
`You can add titles to frames and do other things (morphing and fades) `
That is what I'm now learning.  I have no problem editing video on OpenShot and other Video Editors
Learning editing video running ImageMagick is just a curiosity to me.

Pls advise how to add water mark and/or graphics on images running ImageMagick.  Thanks
Also I have no problem adding text/graphis on GIMP

Regards

**Reply by fmw42** [reply]
https://imagemagick.org/Usage/compose/#watermark

**Reply by satimis** [reply]
Hi Fred,
Thanks for your link

I run following command adding text as watermark on an image.jpg
`convert -size 1400x800 xc:none -pointsize 50 -fill red -gravity NorthWest -draw "text 200,20 'Holliday Resort'" -fill yellow -gravity NorthWest -draw "text 500,400 'Holiday Resort'" miff:-| composite -tile - seychelles500.jpg wmark_seychelles500.jpg`

1) If I want to bold the text and making the text semi-transparent
What will be the command ?

2) If I want to replace the text with a small image.  How to change the command ?

Thanks

![wmark_seychelles500](https://user-images.githubusercontent.com/30752517/210783546-b5159c0b-0ba8-4def-97fb-823426684c1a.jpg)

**Comment by satimis** [comment]
Hi all

I performed following test converting .pdf file direct to video.mp4.  It worked but this is solely a test.  The quality of the output video.mp4 is no good.  In practical run an Open Source Video Editor, such as OpenShot etc to do the job.

The .pdf file was created on image.jpg.  I still have the original images,

Steps:
1)
On Ubuntu 22.04 desktop Terminal run;
`$ convert -density AAA.pdf picture.jpg`

Finally only 8 pictures generated with following warning popup
`convert-im6.q16: cache resources exhausted /tmp/magick-n5Ms7Zl7x2q6EsGKUsxc_ABAmkZc9kOi10 @ error/cache.c/OpenPixelCache/4095.`

There are 30 pages(i.e. 30 pictures) in this .pdf file.

Then on Terminal run;
`$ sudo nano /etc/ImageMagick-6/policy.xml`

On the line: <policy domain="resource" name="disk" value="1GiB"/
change 1GiB to 8GiB

Save and exit the file.  Tried again still having the same result.

2)
Following command works;
`$ convert -density 150 -quality 70 AAA.pdf picture.jpg`

30 pictures generated
```
picture-0.jpg
picture-1.jpg
....
....
picture-28.jpg
picture-29.jpg
```

To convert the picture.jpg to .mp4
On Terminal run;
`$ ffmpeg -r 1/10 -i picture-%01d.jpg -c:v libx264 -r 30 -pix_fmt yuv420p video.mp4`

Finally a video.mp4 is created but it takes long time to complete with following warning pop throughout the process;
`deprecated pixel format used, make sure you did set range correctly`

I just ignore the warning.

However the quality of the video is no good. Besides all picture.jpg are not of equal size therefore some pictures on the video.mp4 being pulled up to meet the size of the video.

It would be better to do it on Open Source editor.

Regards

**Comment by fmw42** [comment]
You have no density value!

`convert -density AAA.pdf picture.jpg`

Use a higher density value than the default 72 then resize back to the original. 72*4=288, so resize by 1/4=25%

`convert -density 288 AAA.pdf -resize 25% picture.jpg`

Or just use a larger density and resize to whatever size you need for your video.

**Reply by satimis** [reply]
Hi Fred,

`convert -density 288 AAA.pdf -resize 25% picture.jpg`
$ convert -density 288 AAA.pdf -resize 25% picture.jpg
Fail with warning
`convert-im6.q16: cache resources exhausted `/tmp/magick-8b2MGOh8T1xeNJWpdKMgRmHpBL0N1McG18' @ error/cache.c/OpenPixelCache/4095.`

/etc/ImageMagick-6/policy.xml
On the line: <policy domain="resource" name="disk" value="1GiB"/
even change -> 1GiB to 16GiB

Regards

**Reply by fmw42** [reply]
You have run out of disk resource.  Your Imagemagick policy.xml has a limit of 1 GB and is suggesting you increase it to 16 GB.

Are you on a shared server or is that your own computer?

How big was your PDF.  Increasing the density is rasterizing it into a rather large image file in both dimensions and file size. That is why this is happening. But your limit of 1 GB seems rather low.

**Reply by satimis** [reply]
`Are you on a shared server or is that your own computer?`
No, this is my own PC with 32G RAM on board

AAA.pdf = 48.5MB

I have tried 16GB limit without result

Following command works (mentioned on my previous posting of yesterday) on this PC
`$ convert -density 150 -quality 70 AAA.pdf picture.jpg`

It generates 30 pictures without complaint.  8GiB limit

Shall I start a new posting about "Semi-Transparent Watermark" which was posted here about 16hrs ago?  Thanks

Regards

**Comment by snibgo** [comment]
> AAA.pdf = 48.5MB

That has no relevance. How many pages in the PDF? How many pixels in each page? (Eg, width * height in inches, multiplied by DPI, multiplied by DPI). That is the total number of pixels. Depending on your IM version, each pixel might take 8 bytes.

**Reply by satimis** [reply]
Hi snibgo,

`How many pages in the PDF?`
30 pages
`
How many pixels in each page?`
Variable, not the same size on each page

Regards
